src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php

```diff
@@ -94,11 +95,23 @@ public function getProxyFactoryCode(Definition $definition, $id, $factoryCode =
      */
     public function getProxyCode(Definition $definition)
     {
-        return preg_replace(
+        $code = $this->classGenerator->generate($this->generateProxyClass($definition));
+
+        $code = preg_replace(
             '/(\$this->initializer[0-9a-f]++) && \1->__invoke\(\$this->(valueHolder[0-9a-f]++), (.*?), \1\);/',
             '$1 && ($1->__invoke(\$$2, $3, $1) || 1) && $this->$2 = \$$2;',
-            $this->classGenerator->generate($this->generateProxyClass($definition))
```

Comment:
you could even keep this code here

Reply:
I could but it’s less symmetric

Reply:
@nicolas-grekas is this replacement related to backporting some ProxyManager too, or would be also keep it after dropping support for ProxyManager 2.1 and older ?

If it is meant to stay, I would not try to make it symmetric with the feature backporting.

Reply:
this is Symfony-only feature, has been refused in ProxyManager (at least not merged yet), so not part of any versions of it